src/System.Net.Security/src/System/Net/SecureProtocols/_SecuritySafeHandles.Windows.cs

```diff
@@ -0,0 +1,1219 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using Microsoft.Win32.SafeHandles;
+using System.Runtime.InteropServices;
+using System.Security.Authentication.ExtendedProtection;
+
+namespace System.Net.Security
+{
+    //
+    // Used when working with SSPI APIs, like SafeSspiAuthDataHandle(). Holds the pointer to the auth data blob.
+    //
+#if DEBUG
+    internal sealed class SafeSspiAuthDataHandle : DebugSafeHandle
+    {
+#else
+    internal sealed class SafeSspiAuthDataHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+#endif
+        public SafeSspiAuthDataHandle() : base(true)
+        {
+        }
+
+        protected override bool ReleaseHandle()
+        {
+            return Interop.Secur32.SspiFreeAuthIdentity(handle) == Interop.SecurityStatus.OK;
+        }
+    }
+
+    //
+    //  A set of Safe Handles that depend on native FreeContextBuffer finalizer
+    //
+#if DEBUG
+    internal abstract class SafeFreeContextBuffer : DebugSafeHandle
+    {
+#else
+    internal abstract class SafeFreeContextBuffer : SafeHandleZeroOrMinusOneIsInvalid
+    {
+#endif
+        protected SafeFreeContextBuffer() : base(true) { }
+
+        // This must be ONLY called from this file and in the context of a CER
```

Comment:
Aren't ConstrainedExecutionRegions a thing of the past?

Also, the method doesn't really need the `unsafe` keyword, it uses IntPtr, not void*.